couchbase/cluster.hxx

```diff
@@ -99,14 +97,16 @@ class cluster
     template<typename Handler>
     void open_bucket(const std::string& bucket_name, Handler&& handler)
     {
-        if (buckets_.find(bucket_name) != buckets_.end()) {
+        auto b = find_bucket_by_name(bucket_name);
+        if (b) {
             return handler({});
         }
+
         std::vector<protocol::hello_feature> known_features;
         if (session_ && session_->has_config()) {
             known_features = session_->supported_features();
         }
-        auto b = std::make_shared<bucket>(id_, ctx_, tls_, tracer_, meter_, bucket_name, origin_, known_features);
+        b = std::make_shared<bucket>(id_, ctx_, tls_, tracer_, meter_, bucket_name, origin_, known_features);
```

Comment:
b is being emplaced into buckets_ without a lock being held.

Reply:
thanks